astropy/visualization/wcsaxes/helpers.py

```diff
@@ -24,6 +28,56 @@
 }
 
 
+def add_compass(
+    ax,
+    north_angle,
+    length=0.15,
+    sep_N=-0.1,
+    sep_E=0.04,
+    color="white",
+    **kwargs,
+):
+    """
+    Display North, East arrows.
+
+    Parameters
+    ----------
+    ax : :class:`~astropy.visualization.wcsaxes.WCSAxes`
+        WCSAxes instance in which the beam shape and size is displayed. The WCS
+        must be celestial.
+    north_angle : `~astropy.coordinates.Angle`
+        Angle between sky North and pixel coordinate x-axis,
+        along tangent line of great circle running through
+        pixel and sky North.
+        This can be computed by :meth:`astropy.wcs.`
+    length : float, optional.
+        The arrow length, default 0.15.
+    sep_N : float, optional.
+        Separation between North arrow and "N" label, default -0.1.
+    sep_E : float, optional.
+        Separation between East arrow and "E" label, default 0.04.
+    color : str, optional.
+        The color. Default "white".
+    kwargs
+        Additional arguments are passed to
+        :class:`~matplotlib.mpl_toolkits.axes_grid1.anchored_artists.AnchoredDirectionArrows`.
+    """
+    arrow = AnchoredDirectionArrows(
```

Comment:
The original reason I though it was important that pixel be an input argument is to handle the case where you have either a very large FOV or are very close to the north pole. The `corner="upper left"` functionality replaces this, but it's not obvious to me how to back out the pixel location: "upper left" just translates to a single integer that can be read by AnchoredDirectionArrows. Going up the inheritance tree, the location is read somewhere within `matplotlib.offsetbox.OffsetBox`, and it appears that the location of the arrow's origin might change depending on, e.g., the requested size of the labels.  This was already a problem in the old implementation because the arrows were being drawn by the artist in the same way, but at least the user could try to get the x,y pixel from which North was computed near to the location of the arrow in the resulting image.

I see a few options:
1. Make the arrow, then try to back out the pixel value at the origin of the arrows, then compute the north pole angle and modify/re-make the arrow. To me this appears to be a lot of work and is clunky, unless someone can suggest a cleaner and easier way; however, it would be satisfying in the sense that it would auto-magically get North right even for large images
2. Make "pixel" an optional argument, so a user could try to match the location in the image where North is computed with the location where the arrows are getting drawn
3. Leave it as-is, i.e. always compute North at (0, 0), add a disclaimer that it won't work for large images and/or near the north pole, and let someone else implement the edge case if they want to

Perhaps you have a different suggestion @astrofrog, just let me know